### DISCUSSION TITLE
Cannot Run tests due to gdal module and found requirements build improper

### DISCUSSION THREAD

**Discussion by dhananjaypai08** [koordinates/kart#606] [Q&A]
Hi there, I am Dhananjay Pai currently in my sophomore year and looking to contribute in this project and take part in GSOC as well. 

After setting up kart on local build and after installing the requirements for test as well as for local development when I try to run any test it shows that there are still some modules that aren't installed. After looking into it I found that the module **osgeo** is creating an issue. I have the GDAL distribution installed on my pc as well as the V3.9py distribution for gdal. But still it gives this message
```
from osgeo import ogr, osr
E   ModuleNotFoundError: No module named 'osgeo'
```
Osgeo is imported in several different files and I can run the tests
Reason for testing was to get to know codeflow and working of kart.
Also there are certain import errors 

```
ImportError while loading conftest 'C:\Users\dhana\coding\gsoc\kart\tests\conftest.py'.
conftest.py:28: in <module>
    from kart.diff_estimation import terminate_estimate_thread
```
There's no function named terminate_estimate_thread in kart/diff_estimation.py 
myself being a beginner here asked it here if someone can help it would be great. Thanks

Also will be looking to propose for kart CLI help improvements having built and developed several cli's under psf sub-org zyte and many more

**Comment by craigds** [comment]
Thanks for introducing yourself. 
[terminate_estimate_thread is defined here](https://github.com/koordinates/kart/blob/master/kart/diff_estimation.py#L65); could you post the full traceback for that error?

Could you tell us what version of GDAL you have installed, and how it was installed?

Also how are you building Kart? Check the instructions in [CONTRIBUTING.md](https://github.com/koordinates/kart/blob/master/CONTRIBUTING.md#building-the-development-version-with-cmake-macos--linux)

**Reply by dhananjaypai08** [reply]
> Thanks for introducing yourself. [terminate_estimate_thread is defined here](https://github.com/koordinates/kart/blob/master/kart/diff_estimation.py#L65); could you post the full traceback for that error?
> 
> Could you tell us what version of GDAL you have installed, and how it was installed?
> 
> Also how are you building Kart? Check the instructions in [CONTRIBUTING.md](https://github.com/koordinates/kart/blob/master/CONTRIBUTING.md#building-the-development-version-with-cmake-macos--linux)

Hey, thank you so much for the clarification. I apologize for not going through contributing building.

GDAL version: GDAL 3.4.2, released 2022/03/08
python version: Python 3.10.4

installation source:  [https://www.gisinternals.com/query.html?content=filelist&file=release-1928-x64-gdal-3-4-2-mapserver-7-6-4.zip]( https://www.gisinternals.com/query.html?content=filelist&file=release-1928-x64-gdal-3-4-2-mapserver-7-6-4.zip)
have installed the general gdal release for core components and the latest gdal python binding on windows(64 bit). Environment Path has been defined for gdal, gdal_plugins.

Input : 
```
kart/tests $python3 -m pytest conftest.py
````

Output:
```
ImportError while loading conftest 'C:\Users\dhananjay\kart\tests\conftest.py'.
conftest.py:28: in <module>
    from kart.diff_estimation import terminate_estimate_thread
..\kart\__init__.py:108: in <module>
    from osgeo import gdal, ogr, osr
E   ModuleNotFoundError: No module named 'osgeo'
```

Thanks alot for answering about the build for local sets will try to do it that way.
